pkg/manager/manager.go

```diff
@@ -142,18 +142,23 @@ type Options struct {
 	LeaderElection bool
 
 	// LeaderElectionResourceLock determines which resource lock to use for leader election,
-	// defaults to "configmapsleases". Change this value only if you know what you are doing.
+	// defaults to "leases". Change this value only if you know what you are doing.
 	// Otherwise, users of your controller might end up with multiple running instances that
 	// each acquired leadership through different resource locks during upgrades and thus
 	// act on the same resources concurrently.
-	// If you want to migrate to the "leases" resource lock, you might do so by migrating to the
-	// respective multilock first ("configmapsleases" or "endpointsleases"), which will acquire a
-	// leader lock on both resources. After all your users have migrated to the multilock, you can
-	// go ahead and migrate to "leases". Please also keep in mind, that users might skip versions
-	// of your controller.
 	//
-	// Note: before controller-runtime version v0.7, the resource lock was set to "configmaps".
-	// Please keep this in mind, when planning a proper migration path for your controller.
+	// If you are using `configmaps`/`endpoints` resource lock and want to migrate to "leases",
+	// you might do so by migrating to the respective multilock first ("configmapsleases" or "endpointsleases"),
+	// which will acquire a leader lock on both resources.
+	// After all your users have migrated to the multilock, you can go ahead and migrate to "leases".
+	// Please also keep in mind, that users might skip versions of your controller.
+	//
+	// Note: before controller-runtime version v0.7, it was set to "configmaps".
+	// And from v0.7 to v0.11, the default was "configmapsleases", which was
+	// used to migrate from configmaps to leases.
+	// Since the default was "configmapsleases" for over a year, spanning five minor releases,
+	// any actively maintained operators are very likely to have a released version that uses
+	// "configmapsleases". Therefore defaulting to "leases" should be safe since v0.12.
```

Comment:
I understand that it is safe only for new projects OR for those that properly migrated.
So, my concern here is that those who have operators and just bump C+R might not be aware and do not know the steps. In this way, wdyt about: 
- We add the steps to `migrating to the respective multilock first ("configmapsleases" or "endpointsleases"),`
- What exactly do users must do? Have we an example?

Reply:
Aha, actually I just optimize the format for this paragraph, you can see the removed lines above. But sure, I'm going to add steps and examples to make the migration more clear.

Reply:
Aha, actually this paragraph is just from the internal comment where the default value is set.

https://github.com/kubernetes-sigs/controller-runtime/blob/3f265c36d7bf597facc9b0363b48ff21a68cb782/pkg/leaderelection/leader_election.go#L60-L66

But sure, I'm going to add steps and examples to make the migration more clear.

Reply:
I think after we have it in the docs we could highlight it in the release page: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.12.0

Reply:
> I think after we have it in the docs we could highlight it in the release page: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.12.0

Yeah, makes sense. I have updated the docs again, please see if I have made it clear.